zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionProcessingState.java

```diff
@@ -69,35 +75,61 @@ public boolean shouldProcess() {
   }
 
   public boolean isExportingPaused() {
-    return isExportingPaused;
+    return exporterState.equals(PAUSED);
   }
 
   @SuppressWarnings({"squid:S899"})
-  /** Returns true if exporting is paused */
-  public boolean pauseExporting() throws IOException {
-    final File persistedExporterPauseState =
-        getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME);
-    persistedExporterPauseState.createNewFile();
-    if (persistedExporterPauseState.exists()) {
-      isExportingPaused = true;
-      return true;
+  /** Returns true if exporting is paused. This method overrides the effects of soft pause. */
+  public boolean pauseExporting() {
+    try {
+      setPersistedExporterState(PAUSED);
+    } catch (final IOException e) {
+      return false;
+    }
+    return true;
+  }
+
+  /** Returns true if soft exporting is paused. This method overrides the effects of hard pause. */
+  public boolean softPauseExporting() {
+    try {
+      setPersistedExporterState(SOFT_PAUSED);
+    } catch (final IOException e) {
+      return false;
     }
-    return false;
+    return true;
   }
 
-  /** Returns true if exporting is resumed */
-  public boolean resumeExporting() throws IOException {
+  /** Returns true if exporting is resumed. This method resumes both soft and "hard" exporting. */
+  public boolean resumeExporting() {
+    try {
+      setPersistedExporterState(EXPORTING);
+    } catch (final IOException e) {
+      return false;
+    }
+    return true;
+  }
+
+  void setPersistedExporterState(final String state) throws IOException {
     final File persistedExporterPauseState =
         getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME);
-    Files.deleteIfExists(persistedExporterPauseState.toPath());
-    if (!persistedExporterPauseState.exists()) {
-      isExportingPaused = false;
-      return true;
-    }
-    return false;
+    Files.write(persistedExporterPauseState.toPath(), Arrays.asList(state), StandardCharsets.UTF_8);
+    exporterState = state;
   }
 
   private void initExportingState() {
-    isExportingPaused = getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME).exists();
+    Stream<String> stream = null;
+    try {
+      if (!getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME).exists()) {
+        setPersistedExporterState(EXPORTING);
+      } else {
+        stream =
+            Files.lines(getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME).toPath());
+        exporterState = stream.findFirst().get();
+        stream.close();
+      }
+    } catch (final IOException e) {
+      // exporting is the default state
+      exporterState = EXPORTING;
+    }
```

Comment:
:x: I think this is not backwards compatible. If we have paused exporter and then do an upgrade, it will resume because the file is empty. We could have to do something like this:

```
   else {
        final var state =
            Files.readString(
                getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME).toPath());
        if (state == null || state.isEmpty() || state.isBlank()) {
          // Backwards compatibility. If the file exists, it is paused.
          setPersistedExporterState(PAUSED);
        }
        exporterState = getEnum(state);
      }
    } 
```